src/Components/Web.JS/src/Services/NavigationEnhancement.ts

```diff
@@ -50,6 +50,12 @@ export function attachProgressivelyEnhancedNavigationListener(callbacks: Navigat
   window.addEventListener('popstate', onPopState);
 
   attachProgrammaticEnhancedNavigationHandler(performProgrammaticEnhancedNavigation);
+
+  if (document.readyState === 'loading') {
+    document.addEventListener('DOMContentLoaded', onInitialDomContentLoaded);
+  } else {
+    onInitialDomContentLoaded();
+  }
```

Comment:
This is based on [this example](https://developer.mozilla.org/docs/Web/API/Document/DOMContentLoaded_event#checking_whether_loading_is_already_complete) from MDN.

I verified on both Edge and Firefox (manually and via a new E2E test) that the document's ready state will be `'loading'` and the `'DOMContentLoaded'` event won't fire until the initial page response completes.

Reply:
This is interesting and likely points to a potential subtle bug in this area.

If the issue is that the document is not complete by the time we execute the Blazor script, (and scan for components) then it might mean that any component prerendered after the `<script>` tag might also suffer from this, and as such, we might need to delay the process until the document has been parsed with `defer` instead.

Reply:
> If the issue is that the document is not complete by the time we execute the Blazor script, (and scan for components) then it might mean that any component prerendered after the <script> tag might also suffer from this, and as such, we might need to delay the process until the document has been parsed with defer instead.

The issue actually wasn't that component comments weren't getting detected by the initial document scan. Rather, components were getting activated for interactivity as soon as they got included in a streaming response, instead of waiting to get activated until the stream response ends. We want to wait until the streaming response ends before activating interactive components, because it already works this way when an enhanced navigation uses streaming rendering (it was a deliberate choice to make new components get activated only after the enhanced navigation fully completes). So this PR just makes the streaming rendering behavior consistent between full page reloads and enhanced navigations.

> and as such, we might need to delay the process until the document has been parsed with defer instead.

According to [MDN](https://developer.mozilla.org/docs/Web/API/Document/DOMContentLoaded_event), the `DOMContentLoaded` event doesn't fire until all deferred scripts have downloaded and executed. So I think we should be good here. The alternative is to use the `load` event, but that waits until unrelated resources (images, stylesheets, etc.) have loaded, so I think that would be a bit too conservative.